06_puppy/mohankrishna/storer_test.go

```diff
@@ -0,0 +1,220 @@
+
+	// then
+	assert.NoError(err, "Should not get an error creating a puppy to a free ID")
+	createdPup, err := s.store.ReadPuppy(0x1236)
+	if assert.NoError(err, "Should be able to read an newly created puppy") {
+		assert.Equal(&newPup, createdPup, "Created puppy should be identical to the one passed to Create")
+	}
+}
+
+func (s *storerSuite) TestMapStoreCreatePuppyIdAlreadyExists() {
+	// given
+	assert := tassert.New(s.T())
+	oldPup := puppy1()
+	newPup := puppy2()
+	newPup.ID = 0x1234
+
+	// when
+	err := s.store.CreatePuppy(&newPup)
+
+	// then
+	assert.Error(err, "Should get an error creating a puppy with existing ID")
+	currentPet, err := s.store.ReadPuppy(0x1234)
+	if assert.NoError(err, "Should be able to read the old existing puppy") {
+		assert.Equal(&oldPup, currentPet, "Created puppy should be identical to the one passed to Create")
+	}
+}
+
+func (s *storerSuite) TestMapStoreUpdatePuppySuccessful() {
+	// given
+	assert := tassert.New(s.T())
+	oldPup := puppy1()
+	updatedPup := puppy2()
+	updatedPup.ID = 0x1234
+
+	// when
+	err := s.store.UpdatePuppy(oldPup.ID, &updatedPup)
+	assert.NoError(err, "Should be able to update the puppy")
+	// then
+	pup, err := s.store.ReadPuppy(oldPup.ID)
+	if assert.NoError(err, "Should be able to read the updated puppy") {
+		assert.Equal(updatedPup, *pup, "Updated puppy should be identical to the one passed to Update")
+	}
+}
+
+func (s *storerSuite) TestMapStoreUpdatePuppyIDDoesnotExist() {
+	// given
+	assert := tassert.New(s.T())
+	updatedPup := puppy2()
+	updatedPup.ID = 0x123
+
+	// when
+	err := s.store.UpdatePuppy(updatedPup.ID, &updatedPup)
+
+	// then
+	assert.NoError(err, "Updating existing puppy should not throw an error")
+	pup, err := s.store.ReadPuppy(updatedPup.ID)
+	if assert.NoError(err, "Updating to non-existing puppy ID is not an error") {
+		assert.Equal(updatedPup, *pup, "Updated puppy should be identical to the one passed to Update")
+	}
+}
+
+func (s *storerSuite) TestMapStoreUpdatePuppyInvalidIDs() {
+	// given
+	assert := tassert.New(s.T())
+	updatedPup := puppy1()
+
+	// when
+	err := s.store.UpdatePuppy(0x123, &updatedPup)
+
+	// then
+	assert.Error(err, "Updating with un matching id and pet.ID is an error")
+}
+
+func (s *storerSuite) TestMapStoreDeletePuppySuccessful() {
+	//given
+	assert := tassert.New(s.T())
+
+	// when
+	deleted, err := s.store.DeletePuppy(0x1234)
+
+	// then
+	assert.NoError(err, "Delete should successfully delete a puppy")
+	assert.True(deleted, "Delete should return true indicating a puppy was deleted")
+
+	_, err = s.store.ReadPuppy(0x1234)
+	assert.Error(err, "Should not be able to read a deleted puppy")
+}
+
+func (s *storerSuite) TestMapStoreDeletePuppyIDDoesNotExist() {
+	//given
+	assert := tassert.New(s.T())
+
+	// when
+	_, err := s.store.DeletePuppy(0x123234)
+
+	// then
+	assert.Error(err, "Should get an error deleting a non existant ID")
+}
+
+func TestStorerImpls(t *testing.T) {
+	mapSuite := storerSuite{
+		impl: maps,
+	}
```

Comment:
Idiomatic go would usually be to have this on one line:
```
    mapSuite := storerSuite{impl: maps}
```

Vertical whitespace can be valuable in making code clearer, but should only be used when necessary (judgement call, of course).